Source/Core/Core/GeckoCode.cpp

```diff
@@ -156,35 +155,31 @@ static bool InstallCodeHandler()
 
 void RunCodeHandler()
 {
-  if (SConfig::GetInstance().bEnableCheats && active_codes.size() > 0)
-  {
-    if (!code_handler_installed || PowerPC::HostRead_U32(INSTALLER_BASE_ADDRESS) - 0xd01f1bad > 5)
-      code_handler_installed = InstallCodeHandler();
+  if (!SConfig::GetInstance().bEnableCheats)
+    return;
 
-    if (!code_handler_installed)
-    {
-      // A warning was already issued.
+  {
+    std::lock_guard<std::mutex> codes_lock(s_active_codes_lock);
+    if (s_active_codes.empty())
       return;
-    }
 
-    if (PC == LR)
+    if (!s_code_handler_installed || PowerPC::HostRead_U32(INSTALLER_BASE_ADDRESS) - 0xd01f1bad > 5)
```

Comment:
Please add a big TODO for the 0xd01f1bad constant. It looks like a dirty hack to me...